tests/test_s3/test_s3.py

```diff
@@ -1943,6 +1943,17 @@ def test_get_stream_gzipped():
     assert res == payload
 
 
+@mock_s3
+def test_duplicated_buckets():
+    s3 = boto3.client('s3', region_name='eu-west-1')
+    s3.create_bucket(Bucket='mybucket')
+
+    try:
+        s3.create_bucket(Bucket='mybucket')
```

Comment:
To actually get the bucket to be created in the non-default region, you need to set the location constraint:
 s3.create_bucket(Bucket='mybucket', CreateBucketConfiguration={'LocationConstraint': 'eu-west-1'})